File: ESTRUTURAS/1110/main.py
from collections import deque


def get_cards(deck):
    discarded = deque()
    while len(deck) > 1:
        discarded.append(deck.popleft())
        deck.rotate(-1)
    return discarded, deck


def print_formatted(discarded, remaining):
    print('Discarded cards: ', end='')
    print(*discarded, sep=', ')
    print('Remaining card:', remaining.pop())


while (n := int(input())):
    deck = deque(range(1, n+1))
    discarded, remaining = get_cards(deck)
    print_formatted(discarded, remaining)

File: ESTRUTURAS/1110/Main.java
import java.util.LinkedList;
import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Main {

    static LinkedList<Integer> getCards(LinkedList<Integer> deck) {
        LinkedList<Integer> discarded = new LinkedList<Integer>();
        while (deck.size() > 1) {
            discarded.add(deck.removeFirst());
            deck.add(deck.removeFirst());
        }
        return discarded;
    }

    static void printFormatted(LinkedList<Integer> discarded, LinkedList<Integer> remaining) {
        System.out.println("Discarded cards: " + discarded.stream().map(String::valueOf).collect(Collectors.joining(", ")));
        System.out.println("Remaining card: " + remaining.getFirst());
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n;
        while ((n = scanner.nextInt()) > 0) {
            LinkedList<Integer> deck = new LinkedList<Integer>(IntStream.rangeClosed(1, n).boxed().toList());
            LinkedList<Integer> discarded = getCards(deck);
            printFormatted(discarded, deck);
        }
        scanner.close();
    }
}
